app/src/main/java/com/pajato/android/gamechat/common/InvitationManager.java

```diff
@@ -17,35 +17,185 @@
 import java.util.List;
+import java.util.Locale;
 import java.util.Map;
 
+import static com.pajato.android.gamechat.chat.model.Message.STANDARD;
+
 /**
- * Provide a fragment to handle the display of the rooms available to the current user.
+ * Handle invitations to groups.
  *
  * @author Paul Michael Reilly
  */
-public enum InvitationManager {
+public enum InvitationManager implements ResultCallback<AppInviteInvitationResult> {
     instance;
 
+    /** Simple POJO to hold the group name and common room key while we process the join */
+    class GroupInviteData {
+        String groupName;
+        String commonRoomKey;
+        boolean addedToAccountJoinList;
+        boolean addedToGroupMemberList;
+        boolean addedToCommRoomMemberList;
+
+        GroupInviteData(String groupName, String commonRoomKey) {
+            this.groupName = groupName;
+            this.commonRoomKey = commonRoomKey;
+            addedToAccountJoinList = false;
+            addedToGroupMemberList = false;
+            addedToCommRoomMemberList = false;
+        }
+
+        boolean isDone() {
+            return addedToAccountJoinList && addedToGroupMemberList && addedToCommRoomMemberList;
+        }
+    }
+
+    // Private constants.
+
+    private static final String APP_CODE = "aq5ca";
+    private static final String PLAY_STORE_LINK = "https://play.google.com/apps/testing/com.pajato.android.gamechat";
+    private static final String APP_PACKAGE_NAME = "com.pajato.android.gamechat";
+    private static final String WEB_LINK = "https://github.com/pajato/GameChat";
+
+    /** The logcat TAG. */
+    private static final String TAG = InvitationManager.class.getSimpleName();
+
+    // Private instance variables
+
+    /** The repository for any messages needed. */
+    private Map<Integer, String> messageMap = new HashMap<>();
+
+    /** Keep track of any outstanding invites to groups */
+    private Map<String, GroupInviteData> mInvitedGroups = new HashMap<>();
+
     // Public instance variables.
 
     /** The map managing group invitations. */
     private Map<String, List<String>> mGroupInviteMap = new HashMap<>();
 
     // Public instance methods.
 
+    /** Initialize the two central panels in the app: chat and game/activity. */
+    public void init(final AppCompatActivity context) {
+        messageMap.clear();
+        messageMap.put(R.string.HasJoinedMessage, context.getString(R.string.HasJoinedMessage));
+    }
+        /** Handle an account state change by updating the navigation drawer header. */
+    @Subscribe
+    public void onAuthenticationChange(final AuthenticationChangeEvent event) {
+        Account account = event != null ? event.account : null;
+        if (account == null) return;
+
+        // Update Firebase
+        boolean accountChanged = false;
+        for (String key : mInvitedGroups.keySet()) {
+            // If the account has already joined, don't add it again!
+            if (account.joinList.contains(key)) {
```

Comment:
Fwiw, I have adopted a newer style in writing conditionals.  I only recently started to embrace these styles and do not offer them up as Pajato Technologies conventions but as something that has caught me eye.  There are two aspects that I found convincing. 1) Using
`if (test) {
    doStuff...;
   return;
}
doOtherStuff...;` has the benefit if giving another four characters in lines.  A nit perhaps, but I see it more and more in code that I read.

2) I've stopped using curly braces for a single statement.  The prevailing use of IDEs make errors that liberal use of curly braces exist to prevent now obsolete.  Removing lots of these unnecessary curly braces make for more concise code.